src/session/content/message_row/photo.rs

```diff
@@ -0,0 +1,160 @@
+use glib::clone;
+use gtk::{gdk, gio, glib, prelude::*, subclass::prelude::*, CompositeTemplate};
+use tdgrand::{enums::MessageContent, types::File};
+
+use crate::session::chat::{BoxedMessageContent, Message};
+use crate::session::content::{message_row::Media, MessageRow, MessageRowExt};
+use crate::utils::parse_formatted_text;
+use crate::Session;
+
+mod imp {
+    use super::*;
+    use glib::WeakRef;
+    use std::cell::RefCell;
+
+    #[derive(Debug, Default, CompositeTemplate)]
+    #[template(resource = "/com/github/melix99/telegrand/ui/content-message-photo.ui")]
+    pub struct MessagePhoto {
+        pub binding: RefCell<Option<gtk::ExpressionWatch>>,
+        pub handler_id: RefCell<Option<glib::SignalHandlerId>>,
+        pub old_message: WeakRef<glib::Object>,
+        #[template_child]
+        pub media: TemplateChild<Media>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for MessagePhoto {
+        const NAME: &'static str = "ContentMessagePhoto";
+        type Type = super::MessagePhoto;
+        type ParentType = MessageRow;
+
+        fn class_init(klass: &mut Self::Class) {
+            Self::bind_template(klass);
+        }
+
+        fn instance_init(obj: &glib::subclass::InitializingObject<Self>) {
+            obj.init_template();
+        }
+    }
+
+    impl ObjectImpl for MessagePhoto {
+        fn constructed(&self, obj: &Self::Type) {
+            self.parent_constructed(obj);
+            obj.connect_message_notify(|obj, _| obj.update_widget());
```

Comment:
I see a pattern here. 
Not relevant for this PR, but maybe we could override the `ObjectImpl` for `MessageBox` with this here
and put a `update_widget` fn in the `MessageBoxExt` trait.
Not sure whether this will work, but we could generalize a bit if it does.

You can close this, I just wanted to point out.

Reply:
Yeah, I noticed this too. That's something to consider changing later.